base/src/main/java/proguard/evaluation/Variables.java

```diff
@@ -221,7 +232,12 @@ public Value load(int index)
      */
     public IntegerValue iload(int index)
     {
-        return load(index).integerValue();
+        try {
+            return load(index).integerValue();
+        } catch (InstructionEvaluationException e) {
+            throw new VariableInstructionEvaluationException("Value in slot "+index+" is not an integer",
+                  "You might have forgotten to load the value first, or maybe you loaded it to another variable, or you stored a wrong type here.");
```

Comment:
The message makes assumptions about how the code is being used. In this case, "You may have forgotten" assumes that the user has manually created the evaluated code. This may become very confusing if the exception is thrown on incorrectly optimized code, for instance. It should be better to catch such exceptions in the calling code and add informed comments there.

Reply:
Right, in that case it might be better (at least for this PR) to not include hints at all since adding them in calling code might increase code complexity too much.